cachix/src/Cachix/Client/Push.hs

```diff
@@ -119,16 +119,19 @@ uploadStorePath ::
   RetryStatus ->
   -- | r is determined by the 'PushStrategy'
   m r
-uploadStorePath clientEnv _store cache cb storePath retrystatus = do
+uploadStorePath clientEnv store cache cb storePath retrystatus = do
   let (storeHash, storeSuffix) = splitStorePath $ toS storePath
       name = pushCacheName cache
-  onAttempt cb retrystatus
   narSizeRef <- liftIO $ newIORef 0
   fileSizeRef <- liftIO $ newIORef 0
   narHashRef <- liftIO $ newIORef ("" :: ByteString)
   fileHashRef <- liftIO $ newIORef ("" :: ByteString)
+  pathinfo <- liftIO $ Store.queryPathInfo store $ toS storePath
```

Comment:
`queryPathInfo` does not tolerate subpaths or symlinks. I think it's taken care of elsewhere but you may want to make sure.